// @pulumi/hcloud/loadBalancerNetwork.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * Provides a Hetzner Cloud Load Balancer Network to represent a private network on a Load Balancer in the Hetzner Cloud.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as hcloud from "@pulumi/hcloud";
 *
 * const lb1 = new hcloud.LoadBalancer("lb1", {
 *     loadBalancerType: "lb11",
 *     networkZone: "eu-central",
 * });
 * const mynet = new hcloud.Network("mynet", {ipRange: "10.0.0.0/8"});
 * const foonet = new hcloud.NetworkSubnet("foonet", {
 *     networkId: mynet.id,
 *     type: "cloud",
 *     networkZone: "eu-central",
 *     ipRange: "10.0.1.0/24",
 * });
 * const srvnetwork = new hcloud.LoadBalancerNetwork("srvnetwork", {
 *     loadBalancerId: lb1.id,
 *     networkId: mynet.id,
 *     ip: "10.0.1.5",
 * });
 * ```
 *
 * ## Import
 *
 * Load Balancer Network entries can be imported using a compound ID with the following format`<load-balancer-id>-<network-id>`
 *
 * ```sh
 *  $ pulumi import hcloud:index/loadBalancerNetwork:LoadBalancerNetwork myloadbalancernetwork 123-654
 * ```
 */
class LoadBalancerNetwork extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["enablePublicInterface"] = state ? state.enablePublicInterface : undefined;
            inputs["ip"] = state ? state.ip : undefined;
            inputs["loadBalancerId"] = state ? state.loadBalancerId : undefined;
            inputs["networkId"] = state ? state.networkId : undefined;
            inputs["subnetId"] = state ? state.subnetId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.loadBalancerId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'loadBalancerId'");
            }
            inputs["enablePublicInterface"] = args ? args.enablePublicInterface : undefined;
            inputs["ip"] = args ? args.ip : undefined;
            inputs["loadBalancerId"] = args ? args.loadBalancerId : undefined;
            inputs["networkId"] = args ? args.networkId : undefined;
            inputs["subnetId"] = args ? args.subnetId : undefined;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(LoadBalancerNetwork.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing LoadBalancerNetwork resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new LoadBalancerNetwork(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LoadBalancerNetwork.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LoadBalancerNetwork.__pulumiType;
    }
}
exports.LoadBalancerNetwork = LoadBalancerNetwork;
/** @internal */
LoadBalancerNetwork.__pulumiType = 'hcloud:index/loadBalancerNetwork:LoadBalancerNetwork';
